tock/api/views.py

```diff
@@ -206,6 +242,29 @@ def get_queryset(self):
 
         return get_user_timecard_count(timecards)
 
+class FullTimecardList(generics.ListAPIView):
+    serializer_class = FullTimecardSerializer
+
+    def get_queryset(self):
+        # Lookup the associated user and reporting_period in the original
+        # query since we'll be accessing them later. See https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.select_related
+        queryset = Timecard.objects.select_related(
+            'user',
+            'reporting_period',
+        )
+        if self.request.query_params.get('only_submitted') is not None:
+            queryset = queryset.filter(submitted=True)
+
+        earliest_reporting_period_start = self.request.query_params.get('earliest_reporting_period_start')
```

Comment:
Wasn't sure if this was the type of parameter that people would find useful - open to changing this and/or adding other query params!

Reply:
Ooo, this is a great spot to re-use some of @neilmb's work from a few months ago -- specifically https://github.com/18F/tock/blob/623554bf1eb490a5e1ef48552063184661c0c339/tock/api/views.py#L228.

Using this will let us filter on a handful of timecard fields, enough for this initial version.

Reply:
Ah good call, refactored to re-use that!